pkg/exclusions/exclusions.go

```diff
@@ -0,0 +1,102 @@
+package exclusions
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/google/osv-scanner/internal/sbom"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+// ExcludePackages checks for matches in a Lockfile's Packages against the list of provided exclusion regex patterns.
+// If a match is found, the package is removed from the Lockfile's Packages.
+// The updated Lockfile is returned.
+func ExcludePackages(r reporter.Reporter, exclusionPatterns []string, parsedLockfile *lockfile.Lockfile) (*lockfile.Lockfile, error) {
+	// Create a new slice of packages to hold the filtered packages.
+	// Filtered packages are packages that do not match any of the exclusion patterns.
+	var filteredPackages []lockfile.PackageDetails
+
+	// Iterate through the packages in the Lockfile and check for matches against the exclusion patterns.
+	for _, pkg := range parsedLockfile.Packages {
+		matched, err := checkForRegexMatch(r, exclusionPatterns, pkg.Name)
+		if err != nil {
+			return parsedLockfile, err
+		}
+		if !matched {
+			filteredPackages = append(filteredPackages, pkg)
+		}
+	}
+	parsedLockfile.Packages = filteredPackages
+	// Return the updated Lockfile.
+	return parsedLockfile, nil
+}
+
+// ExcludeSBOMPackages checks if SBOM's ID matches the list of provided exclusion regex patterns.
+// If a match is found, the SBOM PURL is excluded from the OSV query by returning a boolean value of true to the calling scanSBOMFile function.
+func ExcludeSBOMPackages(r reporter.Reporter, exclusionPatterns []string, id *sbom.Identifier) (bool, error) {
+	matched, err := checkForRegexMatch(r, exclusionPatterns, id.PURL)
+	if err != nil {
+		return false, err
+	}
+	if matched {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// IsRegexPatterns checks if a list of regex patterns are valid regex patterns.
+func IsRegexPatterns(patterns []string) error {
+	for _, pattern := range patterns {
+		isValid := isValidRegex(pattern)
+		if !isValid {
```

Comment:
Merge this with line above. Provides no benefits setting it as a variable.

Reply:
Done